query-node/mappings/src/content/channel.ts

```diff
@@ -544,3 +556,66 @@ export async function content_ChannelFundsWithdrawn({ store, event }: EventConte
 
   await store.save<ChannelFundsWithdrawnEvent>(rewardClaimedEvent)
 }
+
+export async function processChannelPaymentFromMember(
+  store: DatabaseManager,
+  event: SubstrateEvent,
+  memberId: MemberId,
+  message: IMakeChannelPayment,
+  [payeeAccount, amount]: [AccountId32, Balance]
+): Promise<ChannelPaymentMadeEvent> {
+  const member = await getMemberById(store, memberId)
+
+  // Only channel reward accounts are being checked right now as payment destination.
+  // Transfers to any other destination will be ignored by the query node.
+  const channel = await store.get(Channel, { where: { rewardAccount: payeeAccount.toString() } })
+  if (!channel) {
+    inconsistentState('Payment made to unknown channel reward account')
```

Comment:
I agree that using `inconsistentState` in any of the Meta-protocol actions seems an incoherent approach since semantically it signifies inconsistent state in QN (possibly against runtime) which basically is not possible in metaprotocol transactions. However I used it here to end control-flow(since only payments to channels are being processed), as you said error is being caught so based on action success/failure we are saving meta-transaction state in the QN event entity, which by using `invalidMetadata` would not be possible since it just logs the info. 

I found another helper, `unexpectedData` which exactly behaves as `inconsistentState` but does not signifies anything about broken QN state. Should I use that instead of `inconsistentState`?

Reply:
I think ideally you would throw a new, speicifc kind of error like `MetaprotocolError` and then when catching the error you could distinguish between the non-critical `MetaprotocolError` and other errors that should cause the QN to crash, ie.:
```typescript
if (error instanceof MetaprotocolError) {
  /* Handle the non-critical metaprotocol error */
}
else {
  throw error;
}
```
But there are other places where it would need to be fixed too (like comments and reactions etc.)
Perhaps you could make an issue to potentially fix this in the future (in case it's still needed after launching Orion v2)